src/shared/efivars.c

```diff
@@ -412,9 +412,13 @@ int efi_get_boot_option(
 
         if (header->path_len > 0) {
                 uint8_t *dbuf;
-                size_t dnext;
+                size_t dnext, doff;
+
+                doff = offsetof(struct boot_option, title) + title_size;
+                dbuf = buf + doff;
+                if (header->path_len > l - doff)
```

Comment:
i think "doff + header->path_len > l" is more readable

Reply:
Yeah, but it's susceptible to overflow. The present variant is not, because we know for sure that l > doff.